src/conf.c

```diff
@@ -135,8 +135,10 @@ static int validate_common(cfg_t *cfg, const char *provider, int custom)
 	}
 
 	if (!custom && !cfg_getstr(cfg, "username")) {
-		cfg_error(cfg, "Missing username setting for DDNS provider %s", provider);
-		return -1;
+		if (strcmp(provider, "default@freemyip.com") != 0) {	
```

Comment:
The reason for this check is that this provider doesn't require username. And if you don't configure username in inadyn.conf, then inadyn will complain that the username is missing. I added this if() statement to skip this requirement for this provider.
So if I remove this if() statement, then I would need to configure some fake username in the configuration file, which is a bit silly.
Is there a better way to implement it?   Basically, I don't want to configure "username" in the inadyn.conf for this provider.